### ISSUE TITLE
Why is fitness=0 a lot of the time when hyperparameter tuning

### ISSUE THREAD

**Issue by Ggstuddu** [ultralytics/ultralytics#15937]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to use hyperparameter tuning to get a more reasonable set of hyperparameters to train my own dataset, I use the yolov9t model, the code is as follows:

from ultralytics import YOLO
best_model = "yolov9t.pt"
dataset_name='Fridge-Dents_refactor_with_guideline'
if __name__ == '__main__':
    model = YOLO(best_model) # load the best model and hyperparameters
    # Tune hyperparameters on COCO8 for 30 epochs
    model.tune(data="D:/information/study_info/ultralytics-8.2.0/ultralytics/cfg/datasets/{0}.yaml".format(dataset_name),batch=8,imgsz=640,epochs=100,iterations=500,optimizer="AdamW",lr0=0.0001, plots=False, save=False, val=False)

However, what puzzled me was that fitness=0 was found in many iterations during hyperparameter adjustment. I understood that fitness might be low in some iterations due to unreasonable hyperparameters, but it should not be zero.
The fitness.png in tune files is as follows:
![tune_fitness](https://github.com/user-attachments/assets/9f20bfce-8ef5-471b-ab18-3039f934e636)

Also, I've been using the latest version of ultralytics 8.2.84.

So my question is :why fitness=0 so many times during hyperparameter tuning,did I do something wrong?

Thank you for your patience and help.

**Comment by Y-T-G** [comment]
Can you post the output of running `yolo checks` in terminal?

**Comment by Ggstuddu** [comment]
![image](https://github.com/user-attachments/assets/4a0f9a3c-7a1e-4836-92c4-c3a7a0039310)

Thank you for your help!

**Comment by glenn-jocher** [comment]
@Ggstuddu you're welcome! Please run `yolo checks` in your terminal and share the output so we can further diagnose the issue.

**Comment by Ggstuddu** [comment]
@glenn-jocher
![image](https://github.com/user-attachments/assets/c434aecf-4c57-4827-981e-42d9e0605093)
This is the result of running yolo checks in my terminal.
Thank you for your help!

**Comment by Y-T-G** [comment]
It can happen if some of the iterations failed to train.

**Comment by Ggstuddu** [comment]
@Y-T-G What you say makes a lot of sense, but I wonder  why some of the iterations fail to train. Can you list some possible reasons? I'll check it out.

Thank you very much for your help

**Comment by Y-T-G** [comment]
You would have to check the logs if you saved it. When it fails, it will mention the reason in the logs.

**Comment by Ggstuddu** [comment]
After your reminding, I found that there would be corresponding "events.out.tfevents“ after each iteration. I wonder if there is enough information in this file to find out what the problem is.
I used tensorboard to visualize this file when fitness=0 and found that the training loss was normal, but the validation set loss was always zero. I still don't know why this happened.
![image](https://github.com/user-attachments/assets/0c64a899-a4e8-4c63-9755-1d7ef4553368)
I am new in deep learning,i dont konw if i use this "events.out.tfevents“ file corrcetly.

Thank you very much for your help

**Comment by glenn-jocher** [comment]
It seems like the validation set might not be properly configured or loaded. Please ensure your validation data is correctly specified in your dataset YAML file. If the issue persists, try updating to the latest version of the Ultralytics package and rerun your tuning process.

**Comment by Y-T-G** [comment]
The validation metrics would be 0 because validation is not done until the last epoch during tuning to save time.

**Comment by Ggstuddu** [comment]
@glenn-jocher 
I have updated ultralytics to the latest version before hyperparameter tuning. I think I have configured validation set data correctly in data set yaml file, and can load validation set data correctly and calculate fitness in most iterations. And I didn't modify the dataset yaml file during hyperparameter tuning
The contents of my dataset yaml file are shown below：
![image](https://github.com/user-attachments/assets/2b4b4bdf-c4ea-485f-8ba3-26da4780998c)

@Y-T-G 
You're right. I'm sorry. I didn't notice anything so important.
As you can see from the tensorboard image above,this indicates that the validation set data has been successfully loaded the  validation metrics set has been successfully calculated at the last epoch of this iteration.
In this iteration,mAP50 is 58.9, and MAP50-95 is 28.71,I think its fitness should normally be 58.9*0.1+28.71*0.9=31.73, but its final output fitness is 0.Do you know what might have gone wrong?

Thank you very much for your help

**Comment by glenn-jocher** [comment]
@Ggstuddu it appears that the validation metrics are being calculated correctly. The issue might be related to how the fitness score is being computed or logged. Please ensure that the fitness calculation logic in your script aligns with the expected formula. If the problem persists, consider checking for any potential bugs in the fitness logging mechanism.

**Comment by Ggstuddu** [comment]
@glenn-jocher 
I will try the solutions you mentioned and if I have any results I will post them here. 
Thank you again for your patient help

**Comment by glenn-jocher** [comment]
You're welcome! Feel free to update us with your findings. If you encounter any further issues, we're here to help.

**Comment by Y-T-G** [comment]
Fitness would be 0 if any of the steps in the `try` block fails.

https://github.com/ultralytics/ultralytics/blob/main/ultralytics%2Fengine%2Ftuner.py#L191-L202

**Comment by Ggstuddu** [comment]
@Y-T-G  
Thank you for providing me with these clues. I will follow the information you provide to see what went wrong.
I wish you a happy life and all the best.

**Comment by Ggstuddu** [comment]
@Y-T-G 
I feel so sorry to bother you agin.
I started my hyperparameter tunning process yesterday,again,, it appeared that fitness=0 several times.
I found that I post the wrong "events.out.tfevents“ file above(which fitness is not equal zero),and now i post "events.out.tfevents“ file which fitness is 0 as follow(i double check this time).
![20240903105900](https://github.com/user-attachments/assets/1cbfdd5b-7295-4858-835d-42c43bc09e23)
It shows that the training process stopped at the 56th epoch when it should have ended at the 100th epoch.
I remember that you once said that the reason for training suspension can be found through log information, but I cannot find out the reason for training suspension from the "events.out.tfevents "file at present. 
My question is:
How to record the information during training and find out the reason for training interruption?（It would be even better if this could be done through the existing ultralytics framework）
Sorry to waste your time,i really appreciate for your help.

**Comment by Y-T-G** [comment]
You can save the logs during training.

https://askubuntu.com/a/420983

**Comment by Ggstuddu** [comment]
This link helps me a lot.
Thank you again for your patient help!

**Comment by glenn-jocher** [comment]
I'm glad to hear that the information was helpful! If you have any more questions or run into further issues, feel free to ask.

**Comment by Ggstuddu** [comment]
I found the reason why some of the iteration failed to train:
Every iteration that failed to train reported the same error message : permission denied when they tried to write training log files.
The reason why this error occurd : I always use synchronous web disks to synchronize training results, these synchronous web disks will lock files when uploading to the cloud, so in the process of training, it is possible to interrupt training because the log cannot be written.
I feel so sorry to waste your time with my careless mistakes.
Thank you for your patinence,i really appreciate your kindness.

**Comment by glenn-jocher** [comment]
Thank you for identifying the issue and sharing your findings. No problem at all—glad we could assist. If you have any further questions or need additional help, feel free to reach out.
